test/py/test_module_construct.py

```diff
@@ -0,0 +1,68 @@
+import sys
+import migraphx
+try:
+    import numpy as np
+except:
+    sys.exit()
+
+
+def test_add_op():
+    p = migraphx.program()
+    mm = p.get_main_module()
+    param_shape = migraphx.shape(lens=[3, 3], type="float")
+    x = mm.add_parameter("x", param_shape)
+    y = mm.add_parameter("y", param_shape)
+    add_op = mm.add_instruction(migraphx.op("add"), [x, y])
+    mm.add_return([add_op])
+    p.compile(migraphx.get_target("ref"))
+    params = {}
+    params["x"] = params["y"] = np.arange(9).reshape(
+        param_shape.lens()).astype(np.float32)
+    output = p.run(params)[-1]
+    assert (np.array_equal(output, params["x"] + params["y"]))
```

Comment:
The argument class should support the `==` operator. If not we should update that.

Reply:
it already has `__eq__` exposed.